core/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -240,4 +330,87 @@ public void writeTo(StreamOutput out) throws IOException {
     public String toString() {
         return Strings.toString(this);
     }
+
+    public static class InternalSearchResponse {
```

Comment:
It could, it still has the read/write methods. However, they should not be used anywhere else than in SearchResponse, so I'd go one step in the other direction and even reduce the visibility of write/read.